### PR TITLE
[Bug] schedule CQ offset invalid. offset=77, cqMinOffset=0, cqMaxOffset=74, queueId=1

### PR DESCRIPTION
**Pull request by weihubeats**
https://github.com/apache/rocketmq/issues/7077

### CODE REVIEW

**Comment by weihubeats** [comment]
@RongtongJin I understand that it was repaired in this way, take a look

File: broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java [hunk 1]
```diff
@@ -250,7 +250,6 @@ public boolean correctDelayOffset() {
                 }
                 if (correctDelayOffset != currentDelayOffset) {
                     log.error("correct delay offset [ delayLevel {} ] from {} to {}", delayLevel, currentDelayOffset, correctDelayOffset);
-                    offsetTable.put(delayLevel, correctDelayOffset);
```
Comment (RongtongJin) [inline broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java:253]: If it is not put in the table, the correction will have no effect.

Comment (weihubeats) [reply broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java:253]: So there should be no reloading of the `delayOffset.json` file here, right?

File: broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java [hunk 2]
```diff
@@ -218,8 +218,7 @@ public String encode() {
 
     @Override
     public boolean load() {
-        boolean result = super.load();
-        result = result && this.parseDelayLevel();
```
Comment (RongtongJin) [inline broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java:222]: If super.load() is not called, the JSON file will not be parsed, even during startup. IMO, we can add a new method called loadWhenSyncDelayOffset, which will not correct the delay offset when syncDelayOffset is called.

Comment (weihubeats) [reply broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java]: I added a `loadWhenSyncDelayOffset` method, called at `this.brokerController.getScheduleMessageService().loadWhenSyncDelayOffset();`. `loadWhenSyncDelayOffset `doesn't read `delayOffset.json` repeatedly.

Comment (RongtongJin) [reply broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java]: IMO, we should read delayOffset.json and should not correct the delay offset when slave broker sync delay offset.

Comment (weihubeats) [reply broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java]: I don't know if I understand correctly, but we correct the offset when we initialize the startup, not when we synchronize it

Comment (weihubeats) [reply broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java]: This doesn't feel right either.

**Comment by weihubeats** [comment]
Do we not need to call `correctDelayOffset` in syncDelayOffset, because in `syncDelayOffset` we have already called correctDelayOffset on`delayOffset.json` via `MixAll.string2File(delayOffset, fileName);`

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**